infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java

```diff
@@ -0,0 +1,72 @@
+package org.corfudb.infrastructure.remotecorfutable.loglistener.smr;
+
+import com.google.protobuf.ByteString;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import org.corfudb.common.remotecorfutable.RemoteCorfuTableDatabaseEntry;
+import org.corfudb.common.remotecorfutable.RemoteCorfuTableVersionedKey;
+import org.corfudb.infrastructure.remotecorfutable.DatabaseHandler;
+import org.corfudb.runtime.collections.remotecorfutable.RemoteCorfuTableSMRMethods;
+import org.rocksdb.RocksDBException;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * This class contains the logic for the delete operation from stream listener to database.
+ *
+ * Created by nvaishampayan517 on 08/19/21
+ */
+@AllArgsConstructor
+public class DeleteOperation implements SMROperation {
+    @NonNull
+    private final ByteString[] args;
```

Comment:
Why not List of ByteStrings?

Reply:
The amount of args associated with an operation is fixed while reading from the log entry, so I chose to use a final array.